src/validator.ts

```diff
@@ -140,14 +140,14 @@ export namespace RegExpValidator {
          * A function that is called when the validator entered a RegExp literal.
          * @param start The 0-based index of the first character.
          */
-        onLiteralEnter?(start: number): void
+        onLiteralEnter: (start: number) => void
```

Comment:
I think we need to put the `?` back.

```suggestion
        onLiteralEnter?: (start: number) => void
```